ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -30,8 +30,14 @@ object Trees {
     }
   }
 
+  /** Either a `Tree` or a `JSSpread`.
+   *
+   *  This is the type of actual arguments to JS applications.
+   */
+  sealed trait TreeOrJSSpread extends IRNode
```

Comment:
Any reason this is not an abstract class?

Reply:
Only the fact that it doesn't quite make sense. It's basically a union type. If in a later version of the IR, we have to add another such union type (because of a new feature in ES, for example), we would need to make the other one a `trait`, and then we would have a hard time arguing why `TreeOrJSSpread` was an abstract class and not the new one ;)

Reply:
Fair enough :)